### PR TITLE
Migrate NavigationTelemetry to Kotlin

### PR DESCRIPTION
**Pull request by andrlee**
## Description

Migrate to Kotlin [issue](https://github.com/mapbox/navigation-sdks/issues/166)

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

## Testing

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

**Comment by Lebedinsky** [comment]
@andrlee The tests are failed, please 👀

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationTelemetry.kt [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+    private val queuedFeedbackEvents = ArrayList<FeedbackEvent>()
+    private val locationBuffer: RingBuffer<Location> = RingBuffer(LOCATION_BUFFER_MAX_SIZE)
+    private var metricProgress: MetricsRouteProgress = MetricsRouteProgress(null)
+    private var navigationSessionState: SessionState = SessionState.builder().build()
+    private var metricLocation: MetricsLocation = MetricsLocation(null)
+    private val gpsEventFactory = InitialGpsEventFactory()
+    private var isOffRoute: Boolean = false
+    private var isInitialized = false
+
+    private var lifecycleMonitor: NavigationLifecycleMonitor? = null
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationTelemetry.kt]: Could this be `lifecycleMonitor` be non-nullable using `lateinit`?

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationTelemetry.kt [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+    private fun sendRouteRetrievalEventIfExists() {
+        routeRetrievalElapsedTime?.let {
+            routeRetrievalEvent(it, routeRetrievalUuid)
+            routeRetrievalElapsedTime = null
+            routeRetrievalUuid = ""
+        }
+    }
+
+    private fun sendCancelEvent() {
+        if (navigationSessionState.startTimestamp() != null) {
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationTelemetry.kt]: Should we use `ifNonNull` extension here?

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationTelemetry.kt (hunk 1 above)
Comment (Guardiola31337) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationTelemetry.kt]: It's fine as `NavigationTelemetry` could be alive greater time than `NavigationService` and we want to reset the values.

**Pull request merged**